### PR TITLE
fix coverity defects with CID 147613 147614 147616 147617

### PR DESCRIPTION
**Pull request by luozhengzheng**
Fix coveritsy defects:
coverity scan CID:147617,type: resource leaks
coverity scan CID:147616,type: resource leaks
coverity scan CID:147614,type: resource leaks
coverity scan CID:147613,type: resource leaks

thanks.

### CODE REVIEW

**Comment by heary-cao** [comment]
smb.c   smb_retrieve_shares function is update.
https://github.com/zfsonlinux/zfs/pull/5097

**Comment by luozhengzheng** [comment]
@heary-cao  sorry, I did not notice it. Have updated! thanks.

File: tests/zfs-tests/cmd/xattrtest/xattrtest.c [hunk 1]
```diff
@@ -630,6 +631,7 @@ unlink_files(void)
 		rc = unlink(file);
 		if ((rc == -1) && (errno != ENOENT)) {
 			ERROR("Error %d: unlink(%s)\n", errno, file);
+			free(file);
```
Comment (behlendorf) [inline tests/zfs-tests/cmd/xattrtest/xattrtest.c]: File doesn't need to be freed, it wasn't dynamically allocated above.

File: cmd/zpios/zpios_util.c [hunk 2]
```diff
@@ -260,12 +263,13 @@ set_noise(uint64_t *noise, char *optarg, char *arg)
 int
 set_load_params(cmd_args_t *args, char *optarg)
 {
-	char *param, *search, comma[] = ",";
+	char *param, *search, *relief, comma[] = ",";
```
Comment (behlendorf) [inline cmd/zpios/zpios_util.c]: I'd suggest calling this something like `searchdup` instead which is more descriptive.  It's a little long for the usage here that's fine.

File: tests/zfs-tests/cmd/xattrtest/xattrtest.c (hunk 1 above)
Comment (luozhengzheng) [inline tests/zfs-tests/cmd/xattrtest/xattrtest.c]: @behlendorf  
I see file is allocated by malloc above. as below
"
 	file = malloc(PATH_MAX);
 	if (file == NULL) {
 		rc = ENOMEM;
 		ERROR("Error %d: malloc(%d) bytes for file name\n",
 		    rc, PATH_MAX);
 		goto out;
 	}"

File: cmd/zpios/zpios_util.c (hunk 2 above)
Comment (luozhengzheng) [inline cmd/zpios/zpios_util.c]: @behlendorf  thanks for your nice suggestion.

File: tests/zfs-tests/cmd/xattrtest/xattrtest.c (hunk 1 above)
Comment (behlendorf) [inline tests/zfs-tests/cmd/xattrtest/xattrtest.c]: Never mind, you're right.  I got this hunk confused with the `file` variable in `drop_caches()` from the previous hunk.

File: cmd/zpios/zpios_util.c [hunk 3]
```diff
@@ -260,12 +263,13 @@ set_noise(uint64_t *noise, char *optarg, char *arg)
...
 set_load_params(cmd_args_t *args, char *optarg)
 {
-	char *param, *search, comma[] = ",";
+	char *param, *search, *searchdup, comma[] = ",";
 	int rc = 0;
 
 	search = strdup(optarg);
 	if (search == NULL)
 		return (ENOMEM);
+	relief = search;
```
Comment (behlendorf) [inline cmd/zpios/zpios_util.c]: You're gonna want to get this one too.

Comment (luozhengzheng) [inline cmd/zpios/zpios_util.c]: @behlendorf  sorry, I forget this. I have updated it now.

**Pull request merged**